### PR TITLE
Make DE 440(s) solar system ephemeris available

### LINKED ISSUE #11544: Add support for the DE440 JPL ephemeris
There is a new JPL ephemeris (DE440, [Park et al. 2021](https://iopscience.iop.org/article/10.3847/1538-3881/abd414)), and JPL HORIZONS has been updated to use it as of today.  `astropy.coordinates.solar_system_ephemeris` should be updated to enable access to DE440 ([114 MB](https://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/de440.bsp)) and DE440s ([31 MB](https://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/de440s.bsp)), in addition to the already-supported DE430 and DE432s.

### PR DESCRIPTION
**Pull request by mhvk**
This ephemeris is the latest and presumably best. Right now, I stated that this would become the default in 5.0, but perhaps that should be 4.3?  It is not a difficult change, but a lot of substitutions in our tests, so I thought I'd better ask before doing the work...

fixes #11544

### CODE REVIEW

**Comment by pllim** [comment]
Does this mean we can revert #11564 or not?

File: docs/changes/coordinates/11601.other.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The new DE440 and DE440s ephemerides are now available via shortcuts
+'de440' and 'de440s'.  The DE 440 ephemeris will become the default in 5.0.
```
Comment (ayshih) [inline docs/changes/coordinates/11601.other.rst:2]: You should clarify that these are JPL ephemerides, and by "default" you mean that DE440 will become the default _JPL_ ephemeris, not the overall default ephemeris (i.e., over the built-in ephemeris).

**Comment by adrn** [comment]
Nice! Indeed a simple update. I don't have strong feelings about making the default change now (for v4.3) vs. v5.0.

**Review by adrn — COMMENTED** [review]
Nice! Indeed a simple update. I don't have strong feelings about making the default change now (for v4.3) vs. v5.0.

File: docs/changes/coordinates/11601.other.rst (hunk 1 above)
Comment (mhvk) [reply docs/changes/coordinates/11601.other.rst]: Good point, done!

**Comment by mhvk** [comment]
OK, I updated the change-log fragment.

I had a quick look at trying to make `de440s` the default, but found rather many tests started failing (e.g., against the built-in ERFA ephemeris). So, probably best to push that change off and keep this simple.

**Comment by mhvk** [comment]
OK, tests passed. Once in, I'll put up a PR to change the default, just to show what changes are needed. That will be for 5.0, though.

**Review by adrn — APPROVED** [review]
Great LGTM, thanks!

**Comment by adrn** [comment]
Was about to hit merge, but then wondered: Do we want to add a test to make sure the new options work? I guess that would mean more remote-data...

**Comment by mhvk** [comment]
@adrn - should have mentioned that - I indeed decided that checking locally was good enough in this case (which I did...). I'm happy at some point to add a test, but then we should use the cache prefill option that now exist so that we're not hammering the JPL servers...

**Comment by mhvk** [comment]
Will do! (But will merge this first so I can have the draft PR with test changes out there too)

**Comment by mhvk** [comment]
See #11609

**Pull request merged**